types/datum.go

```diff
@@ -430,11 +430,9 @@ func (d *Datum) GetValue() interface{} {
 
 // SetValueWithDefaultCollation sets any kind of value.
 func (d *Datum) SetValueWithDefaultCollation(val interface{}) {
-	if val == nil {
-		d.SetNull()
-		return
-	}
```

Comment:
Should we also keep these lines?
If the `val` is nil, would type assertion on it cause panic?

Reply:
It won't cause panic, as [example](https://play.golang.org/p/M44mnWePb6C).